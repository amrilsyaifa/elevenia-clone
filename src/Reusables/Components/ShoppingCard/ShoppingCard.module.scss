@import 'src/Reusables/Styles/Variables';

.card {
  width: 250px;
  min-height: 500px;
  cursor: pointer;
  position: relative;
  .button {
    position: absolute;
    width: 100%;
    bottom: 20px;
    button {
      width: 200px;
    }
  }
}
.wrapper-title {
  display: flex;
  flex-direction: column;
  min-height: 150px;
  .title {
    font-weight: bold;
    margin-top: 20px;
    display: inline;
    -webkit-line-clamp: 2;
    text-overflow: ellipsis;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    word-wrap: break-word;
  }
  .percentage {
    font-size: 12px !important;
    font-weight: normal;
    display: inline;
    -webkit-line-clamp: 2;
    text-overflow: ellipsis;
    overflow: hidden;
    display: -webkit-box;
    -webkit-box-orient: vertical;
    word-wrap: break-word;
  }
  .description {
    margin-top: 5px;
    font-weight: bold;
    font-size: 14px;
  }
}

.wrapper-price {
  display: flex;
  flex-direction: column;
  .old-price {
    font-size: 12px;
    text-decoration: line-through;
    color: $grey;
  }
  .price {
    font-size: 14px;
    font-weight: bold;
    color: $blue;
  }
}

.row {
  margin-top: 10px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  .wrapper-location {
    display: flex;
    flex-direction: column;

    font-size: 13px;
    .name {
      padding-left: 5px;
    }
  }
}
