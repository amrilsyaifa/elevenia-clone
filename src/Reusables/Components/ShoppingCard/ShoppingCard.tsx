import { Typography, Card, Image, Button } from 'antd';
import React from 'react';
import { EnvironmentOutlined, StockOutlined } from '@ant-design/icons';
import Style from './ShoppingCard.module.scss';
import { formatCurrency } from 'src/Reusables/Helpers/CurrencyHelper';

const { Text } = Typography;

interface IShoppingCard {
  isLoading?: boolean;
  url?: string;
  title: string;
  sellerItem: string;
  price: number | string;
  stock: string | number;
  sellCount: string | number;
  ProductOptionDetails?: string;
  onAddToCart?: (e) => void;
  onClick: (e) => void;
}

const ShoppingCard: React.FC<IShoppingCard> = ({
  isLoading,
  title,
  sellerItem,
  price,
  stock,
  sellCount,
  url,
  ProductOptionDetails,
  onClick,
  onAddToCart
}) => {
  return (
    <>
      <Card className={Style['card']} loading={isLoading} onClick={onClick} data-testid="click">
        <Image
          width={200}
          height={200}
          src="error"
          fallback={url || '/assets/images/default-image.jpeg'}
        />
        <div className={Style['wrapper-title']}>
          <div className={Style['title']}>
            <Text>{title}</Text>
          </div>
          <Text className={Style['percentage']}>SKU: {sellerItem || '-'}</Text>
          <Text className={Style['description']}>Deskripsi: </Text>
          <div>{ProductOptionDetails || '-'}</div>
        </div>
        <div className={Style['wrapper-cost']}></div>
        <div className={Style['row']}>
          <div className={Style['wrapper-location']}>
            <div>
              <StockOutlined />
              <Text className={Style['name']}>Stok: {stock}</Text>
            </div>
            <div>
              <EnvironmentOutlined />
              <Text className={Style['name']}>Terjual: {sellCount}</Text>
            </div>
          </div>
          <div className={Style['wrapper-price']}>
            <Text className={Style['price']} data-testid="currency">
              {formatCurrency(+price)}
            </Text>
          </div>
        </div>
        <div className={Style['button']}>
          <Button type="primary" onClick={onAddToCart} data-testid="add">
            Add to Chart
          </Button>
        </div>
      </Card>
    </>
  );
};

export default ShoppingCard;
