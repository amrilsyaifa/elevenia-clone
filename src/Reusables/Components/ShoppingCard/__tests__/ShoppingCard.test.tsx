import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';
import ShoppingCard from '../ShoppingCard';

describe('ShoppingCard', () => {
  test('ShoppingCard Render with empty data', () => {
    const onClick = jest.fn();
    const { container } = render(
      <ShoppingCard
        onClick={onClick}
        isLoading={false}
        title={''}
        sellerItem=""
        stock={0}
        price={0}
        sellCount={''}
      />
    );
    const getTitleText = screen.queryByText('Title');
    expect(getTitleText).toBeNull();
    expect(container).toMatchSnapshot();
  });
  test('ShoppingCard Render with loading true', () => {
    const onClick = jest.fn();
    const { container } = render(
      <ShoppingCard
        isLoading={true}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={5000}
        sellCount={100}
        onClick={onClick}
      />
    );
    const getTitleText = screen.queryByText('Title');
    expect(getTitleText).toBeNull();
    expect(container).toMatchSnapshot();
  });
  test('ShoppingCard Render with loading false', () => {
    const onClick = jest.fn();
    const { container } = render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={5000}
        sellCount={100}
        onClick={onClick}
      />
    );
    const getTitleText = screen.queryByText('Title');
    expect(getTitleText).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test('ShoppingCard Render have sellerItem value', () => {
    const onClick = jest.fn();
    render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem="seller item"
        stock={10000}
        price={5000}
        sellCount={100}
        onClick={onClick}
      />
    );
    const getSellerItemText = screen.getByText('SKU: seller item');
    expect(getSellerItemText).toBeInTheDocument();
  });
  test('ShoppingCard Render have stock value', () => {
    const onClick = jest.fn();
    render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={5000}
        sellCount={100}
        onClick={onClick}
      />
    );
    const getStockText = screen.queryByText('Stok: 10000');
    expect(getStockText).toBeInTheDocument();
  });

  test('ShoppingCard Render have sellCount value', () => {
    const onClick = jest.fn();
    render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={5000}
        sellCount={100}
        onClick={onClick}
      />
    );
    const getSellCountText = screen.queryByText('Terjual: 100');
    expect(getSellCountText).toBeInTheDocument();
  });
  test('ShoppingCard Render have ProductOptionDetails value', () => {
    const onClick = jest.fn();
    render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={5000}
        sellCount={100}
        onClick={onClick}
        ProductOptionDetails="This Description"
      />
    );
    const getProductOptionDetailsText = screen.queryByText('This Description');
    expect(getProductOptionDetailsText).toBeInTheDocument();
  });
  test('ShoppingCard Render have price value', () => {
    const onClick = jest.fn();
    const currency = '5000';
    render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={currency}
        sellCount={100}
        onClick={onClick}
      />
    );
    const getPriceText = screen.getByTestId('currency');
    expect(getPriceText).toBeInTheDocument();
  });
  test('ShoppingCard Render OnClick call', () => {
    const onClick = jest.fn();
    const currency = '5000';
    render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={currency}
        sellCount={100}
        onClick={onClick}
      />
    );
    const pressButton = screen.getByTestId('click');
    fireEvent.click(pressButton);
    expect(onClick).toBeCalled();
  });
  test('ShoppingCard Render onAddToCart call', () => {
    const onClick = jest.fn();
    const onAddToCart = jest.fn();
    const currency = '5000';
    render(
      <ShoppingCard
        isLoading={false}
        title={'Title'}
        sellerItem=""
        stock={10000}
        price={currency}
        sellCount={100}
        onClick={onClick}
        onAddToCart={onAddToCart}
      />
    );
    const pressButton = screen.getByTestId('add');
    fireEvent.click(pressButton);
    expect(onAddToCart).toBeCalled();
  });
});
