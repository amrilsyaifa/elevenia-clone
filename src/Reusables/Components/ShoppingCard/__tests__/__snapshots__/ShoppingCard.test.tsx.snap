// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCard ShoppingCard Render with empty data 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered card"
    data-testid="click"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-image"
        style="width: 200px; height: 200px;"
      >
        <img
          class="ant-image-img"
          src="error"
          style="height: 200px;"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
      <div
        class="wrapper-title"
      >
        <div
          class="title"
        >
          <span
            class="ant-typography"
          />
        </div>
        <span
          class="ant-typography percentage"
        >
          SKU: 
          -
        </span>
        <span
          class="ant-typography description"
        >
          Deskripsi: 
        </span>
        <div>
          -
        </div>
      </div>
      <div
        class="wrapper-cost"
      />
      <div
        class="row"
      >
        <div
          class="wrapper-location"
        >
          <div>
            <span
              aria-label="stock"
              class="anticon anticon-stock"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="stock"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M904 747H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM165.7 621.8l39.7 39.5c3.1 3.1 8.2 3.1 11.3 0l234.7-233.9 97.6 97.3a32.11 32.11 0 0045.2 0l264.2-263.2c3.1-3.1 3.1-8.2 0-11.3l-39.7-39.6a8.03 8.03 0 00-11.3 0l-235.7 235-97.7-97.3a32.11 32.11 0 00-45.2 0L165.7 610.5a7.94 7.94 0 000 11.3z"
                />
              </svg>
            </span>
            <span
              class="ant-typography name"
            >
              Stok: 
              0
            </span>
          </div>
          <div>
            <span
              aria-label="environment"
              class="anticon anticon-environment"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="environment"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                />
              </svg>
            </span>
            <span
              class="ant-typography name"
            >
              Terjual: 
            </span>
          </div>
        </div>
        <div
          class="wrapper-price"
        >
          <span
            class="ant-typography price"
            data-testid="currency"
          >
            RpÂ 0
          </span>
        </div>
      </div>
      <div
        class="button"
      >
        <button
          class="ant-btn ant-btn-primary"
          data-testid="add"
          type="button"
        >
          <span>
            Add to Chart
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ShoppingCard ShoppingCard Render with loading false 1`] = `
<div>
  <div
    class="ant-card ant-card-bordered card"
    data-testid="click"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-image"
        style="width: 200px; height: 200px;"
      >
        <img
          class="ant-image-img"
          src="error"
          style="height: 200px;"
        />
        <div
          class="ant-image-mask"
        >
          <div
            class="ant-image-mask-info"
          >
            <span
              aria-label="eye"
              class="anticon anticon-eye"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="eye"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M942.2 486.2C847.4 286.5 704.1 186 512 186c-192.2 0-335.4 100.5-430.2 300.3a60.3 60.3 0 000 51.5C176.6 737.5 319.9 838 512 838c192.2 0 335.4-100.5 430.2-300.3 7.7-16.2 7.7-35 0-51.5zM512 766c-161.3 0-279.4-81.8-362.7-254C232.6 339.8 350.7 258 512 258c161.3 0 279.4 81.8 362.7 254C791.5 684.2 673.4 766 512 766zm-4-430c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm0 288c-61.9 0-112-50.1-112-112s50.1-112 112-112 112 50.1 112 112-50.1 112-112 112z"
                />
              </svg>
            </span>
            Preview
          </div>
        </div>
      </div>
      <div
        class="wrapper-title"
      >
        <div
          class="title"
        >
          <span
            class="ant-typography"
          >
            Title
          </span>
        </div>
        <span
          class="ant-typography percentage"
        >
          SKU: 
          -
        </span>
        <span
          class="ant-typography description"
        >
          Deskripsi: 
        </span>
        <div>
          -
        </div>
      </div>
      <div
        class="wrapper-cost"
      />
      <div
        class="row"
      >
        <div
          class="wrapper-location"
        >
          <div>
            <span
              aria-label="stock"
              class="anticon anticon-stock"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="stock"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M904 747H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM165.7 621.8l39.7 39.5c3.1 3.1 8.2 3.1 11.3 0l234.7-233.9 97.6 97.3a32.11 32.11 0 0045.2 0l264.2-263.2c3.1-3.1 3.1-8.2 0-11.3l-39.7-39.6a8.03 8.03 0 00-11.3 0l-235.7 235-97.7-97.3a32.11 32.11 0 00-45.2 0L165.7 610.5a7.94 7.94 0 000 11.3z"
                />
              </svg>
            </span>
            <span
              class="ant-typography name"
            >
              Stok: 
              10000
            </span>
          </div>
          <div>
            <span
              aria-label="environment"
              class="anticon anticon-environment"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="environment"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M854.6 289.1a362.49 362.49 0 00-79.9-115.7 370.83 370.83 0 00-118.2-77.8C610.7 76.6 562.1 67 512 67c-50.1 0-98.7 9.6-144.5 28.5-44.3 18.3-84 44.5-118.2 77.8A363.6 363.6 0 00169.4 289c-19.5 45-29.4 92.8-29.4 142 0 70.6 16.9 140.9 50.1 208.7 26.7 54.5 64 107.6 111 158.1 80.3 86.2 164.5 138.9 188.4 153a43.9 43.9 0 0022.4 6.1c7.8 0 15.5-2 22.4-6.1 23.9-14.1 108.1-66.8 188.4-153 47-50.4 84.3-103.6 111-158.1C867.1 572 884 501.8 884 431.1c0-49.2-9.9-97-29.4-142zM512 880.2c-65.9-41.9-300-207.8-300-449.1 0-77.9 31.1-151.1 87.6-206.3C356.3 169.5 431.7 139 512 139s155.7 30.5 212.4 85.9C780.9 280 812 353.2 812 431.1c0 241.3-234.1 407.2-300 449.1zm0-617.2c-97.2 0-176 78.8-176 176s78.8 176 176 176 176-78.8 176-176-78.8-176-176-176zm79.2 255.2A111.6 111.6 0 01512 551c-29.9 0-58-11.7-79.2-32.8A111.6 111.6 0 01400 439c0-29.9 11.7-58 32.8-79.2C454 338.6 482.1 327 512 327c29.9 0 58 11.6 79.2 32.8C612.4 381 624 409.1 624 439c0 29.9-11.6 58-32.8 79.2z"
                />
              </svg>
            </span>
            <span
              class="ant-typography name"
            >
              Terjual: 
              100
            </span>
          </div>
        </div>
        <div
          class="wrapper-price"
        >
          <span
            class="ant-typography price"
            data-testid="currency"
          >
            RpÂ 5.000
          </span>
        </div>
      </div>
      <div
        class="button"
      >
        <button
          class="ant-btn ant-btn-primary"
          data-testid="add"
          type="button"
        >
          <span>
            Add to Chart
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ShoppingCard ShoppingCard Render with loading true 1`] = `
<div>
  <div
    class="ant-card ant-card-loading ant-card-bordered card"
    data-testid="click"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-skeleton ant-skeleton-active"
      >
        <div
          class="ant-skeleton-content"
        >
          <ul
            class="ant-skeleton-paragraph"
          >
            <li />
            <li />
            <li />
            <li
              style="width: 61%;"
            />
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
